required contact fields

amazingphilippe opened the pull request:
Fixes #1873 

# Description

Forces the user to include contact information in non-anonymous report. Removes the skippable contact info.

# Required followup work

Review the quality of reports with these new changes. 

# Checklist:

- [x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [x] I have added and needed tests for my changes (in particular for new screens)
- [x] I have added a comment to any confusing code
- [x] I have added documentation to any modified front-end code. (Or added missing documentation)

DianeLiu2019 commented:
I tested it . It seems no error message shown when the full name field is empty. is that what you want?

amazingphilippe commented:
> full name can still be blank if user go back from summary page and remove their names

I think I fixed it. We should also check with any other validation we have. In this case, when returning from summary page, deleting the name also deleted the key. So I had to check if the key existed in the values when I validated, THEN if they had a non-empty value : `const email = 'email' in values && containsData(values.email)`
